To: vim_dev@googlegroups.com
Subject: Patch 7.3.110
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.110
Problem:    The "nbsp" item in 'listchars' isn't used for ":list".
Solution:   Make it work. (Christian Brabandt)
Files:	    src/message.c


*** ../vim-7.3.109/src/message.c	2011-01-17 20:08:03.000000000 +0100
--- src/message.c	2011-02-01 17:06:21.000000000 +0100
***************
*** 1637,1644 ****
  	else if (has_mbyte && (l = (*mb_ptr2len)(s)) > 1)
  	{
  	    col += (*mb_ptr2cells)(s);
! 	    mch_memmove(buf, s, (size_t)l);
! 	    buf[l] = NUL;
  	    msg_puts(buf);
  	    s += l;
  	    continue;
--- 1637,1652 ----
  	else if (has_mbyte && (l = (*mb_ptr2len)(s)) > 1)
  	{
  	    col += (*mb_ptr2cells)(s);
! 	    if (lcs_nbsp != NUL && list && mb_ptr2char(s) == 160)
! 	    {
! 		mb_char2bytes(lcs_nbsp, buf);
! 		buf[(*mb_ptr2len)(buf)] = NUL;
! 	    }
! 	    else
! 	    {
! 		mch_memmove(buf, s, (size_t)l);
! 		buf[l] = NUL;
! 	    }
  	    msg_puts(buf);
  	    s += l;
  	    continue;
***************
*** 1664,1669 ****
--- 1672,1682 ----
  		    attr = hl_attr(HLF_8);
  		}
  	    }
+ 	    else if (c == 160 && list && lcs_nbsp != NUL)
+ 	    {
+ 		c = lcs_nbsp;
+ 		attr = hl_attr(HLF_8);
+ 	    }
  	    else if (c == NUL && list && lcs_eol != NUL)
  	    {
  		p_extra = (char_u *)"";
*** ../vim-7.3.109/src/version.c	2011-02-01 13:59:44.000000000 +0100
--- src/version.c	2011-02-01 17:08:31.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     110,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
176. You lie, even to user-friends, about how long you were online yesterday.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
